First Course; Soups, Salads, Sandwiches! Splits food.dmi in preparation for future PRs.

Mitchs98 opened the pull request:
## What Does This PR Do
Ports 13 different burgers/sandwiches, 14 different stews/soups, and 9 different salads from [TG](https://github.com/tgstation/tgstation/). Adding credits as I can find them for the food stuffs, some of them I am unable to dig up.
Changes the Super Big Bite Burger to use a Tomato Slice instead of a whole Tomato. 
Renames the Killer Tomato slice to be tomato meat slice
Adds our own version of Pretty Patties - Sprited by ATP Engineer

Splits nearly every food sprite from Food.dmi to be it's own categorized .dmi file, of course properly pathing on each individual food item. This precedent is already set by seafood.dmi and pizza.dmi, so I figure why not follow up on that?

Repaths Burgers and Salads for inheritance/cleaner code.
<details>
<summary>Many, Many PR credits</summary>

https://github.com/tgstation/tgstation/pull/69355 - Salads & Cullen Skink- MidoriWroth

https://github.com/tgstation/tgstation/pull/71914 - Philly Cheesesteak and some Salads + Soups - MidoriWroth

https://github.com/tgstation/tgstation/pull/62031 - BLT - SomethingFish

https://github.com/tgstation/tgstation/pull/53812 - Crazy Burger - SgtHunk

https://github.com/tgstation/tgstation/pull/68390 - PB&J & Peanut Butter Banana - MidoriWroth

https://github.com/tgstation/tgstation/pull/26842 - Eyeball Soup - harmonyn

https://github.com/tgstation/tgstation/pull/45023 - French Onion Soup - harmonyn

https://github.com/tgstation/tgstation/pull/65848 - Zurek - MidoriWroth
</details>

<details>

<summary>Burgers/Sandwiches Recipes</summary>

### Burgers

Crazy Burger - 1 Bun, 2 Meat, 2 Cheese Wedge, 1 Chili, 1 Cabbage, 1 Green Crayon, 1 Flare, 15u Corn Oil
Empowered Burger - 1 Bun, 2 Plasma Sheets
Pretty Patties - 1 Bun, 1 Meat, 5u of Grape Juice/Berry Juice/Lime Juice/Lemon Juice/Orange Juice or 5u Cherry Jelly or 5u Sugar(for White)
Ratburger - 1 Bun, 1 Dead Mouse

### Sandwiches

BLT - 2 Bread, 1 Cabbage, 1 Tomato Slice, 2 Bacon.
Peanut Butter and Jelly Sandwich - 2 Bread, 5u Cherry Jelly OR 5u Slime Jelly, 5u Peanut Butter
Peanut Butter and Banana Sandwich - 2 Bread, 5u Peanut Butter, 1 Banana
Philly Cheesesteak - 2 Bread, 2 Cutlets, 1 Cheese Wedge, 1 Onion

</details>

<details>

<summary>Soup & Stew Recipes</summary>

### Soup

Eyeball Soup - 1 Eyeball, 1 Tomato, 10u Water
Sweet Potato Soup - 10u Water, 10u Milk, 2 Sweet Potatos
Red Beet Soup - 10u Water, 1 Red Beet, 1 Cabbage
French Onion Soup - 10u Water, 1 Onion, 1 Cheese Wedge
Zurek - 10u Water, 10u Flour, 1 Boiled Egg, 1 Cutlet, 1 Carrot, 1 Onion
Cullen Skink - 10u Water, 10u Milk, 4u Black Pepper, 1 Salmon Filet, 1 Onion, 1 Potato
Chicken Noodle Soup - 10u Water, 1 Carrot, 1 Meat, 1 Boiled Spaghetti
Corn Chowder - 10u Water, 5u Cream, 1 Corn, 1 Potato, 1 Carrot, 1 Bacon
Meatball Noodle Soup - 2 Raw Cutlets, 1 Onion, 1 Spaghetti, 2 Meatballs, 1 Peanut
Misklmæsch (Seed Soup) - 10u Water, 10u Vinegar, 1 Pack of Sunflower Seeds, 1 Pack of Lily Seeds, 1 Pack of Ambrosia Vulgaris Seeds


###Stew

Prickeldröndolhaskl (Pickled Bean Stew) - 1 Can of Beans, 1 Cabbage, 1 Tomato, 1 Onion, 1 Chili, 1 Corn
Häfmisklhaskl (Oat Stew) - 1 Oat, 1 Sweet Potato, 1 Parsnip, 1 Carrot
Hong Kong Borscht - 10u Water, 5u Soy Sauce, 1 Tomato, 1 Cabbage, 1 Onion, 1 Carrot, 1 Cutlet
Hong Kong Macaroni Soup - 10u Water, 10u Cream, 1 Boiled Spaghetti, 1 Meat, 1 Bacon
</details>

<details>

<summary>Salad Recipes</summary>

Anti-Pasto Salad - 2 Cabbage, 1 Olive, 1 Tomato, 1 Cutlet, 1 Cheese Wedge
Caesar Salad - 1 Cabbage, 1 Red Onion Slice, 1 Cheese Wedge, 5u Olive Oil, 1 Bread Slice
Citrus Delight - 1 Orange, 1 Lemon, 1 Lime
Fruit Salad - 1 Orange, 1 Apple, 1 Grapes, 2 Watermelon Slices
Greek Salad - 1 Olive, 1 Tomato, 2 Red Onion Slice, 2 Cheese Wedge, 5u Olive Oil
Jungle Salad - 1 Apple, 1 Grapes, 2 Banana, 2 Watermelon Slice
Kale Salad - 1 Carrot, 1 Cabbage, 2 Red Onion Slices, 5u Olive Oil
Potato Salad - 2 Potato, 2 Boiled Eggs, 1 Onion, 5u Mayonnaise
Melon Fruit Bowl - 1 Watermelon, 1 Apple, 1 Orange, 1 Lemon, 1 Banana, 1 Ambrosia
Salad - 2 Cabbages

</details>

## Why It's Good For The Game
A larger variety of food for the chef to prepare is always good, extra fluff for the game. Makes use of recently added condiments and ingredients and gives some uses to lesser used foodstuffs. Also clarity between tomato meat and tomato slices is also quite nice to have to avoid any mix-ups.

## Images of changes
![dreamseeker_9hy1R2j1kU](https://github.com/ParadiseSS13/Paradise/assets/12178527/616e4017-f281-4aeb-aa08-143fbac3341a)
![dreamseeker_bkhUMPqYR1](https://github.com/ParadiseSS13/Paradise/assets/12178527/3ee989c2-9ca3-436f-b8ba-ada00041586a)
![dreamseeker_NqmPwJeRFI](https://github.com/ParadiseSS13/Paradise/assets/12178527/9f7baebe-1851-41aa-9e86-4ed21984cfd2)
![2328024a7eeb2166c718c173465e76f8](https://github.com/ParadiseSS13/Paradise/assets/12178527/24bdfe0f-be59-4ae2-9aaa-efcd89e40085)



## Testing
1. Spawned everything in.
2. Cooked all of everything

## Changelog
:cl:Mitchs98, MidoriWroth, SomethingFish, SgtHunk, harmonyn, ATP Engineer
add: Ports 12 different burgers/sandwiches, 14 different stews/soups, and 9 different salads from TG. Check the wiki or the PR for recipes!
add: Pretty Patties, sprited by ATP Engineer.
add: Salad. Plain, generic salad. 
tweak: Super Big Bite Burger now uses 1 Tomato Slice instead of 1 Tomato.
tweak: The Tomato Slice from Killer Tomato's are now named tomato meat slice for clarity.
refactor: Food.dmi has largely been split into multiple .dmi files, this shouldn't affect players or the game directly in any capacity.
refactor: Burgers and Salads are now their own proper paths for inheritance/cleaner code.
/:cl:

Contrabang commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -136,6 +136,166 @@
+	filling_color = "#808080"
+	bitesize = 2
+	list_reagents = list("nutriment"= 5, "protein" = 1)
+	tastes = list("bun" = 2, "dead rat" = 5)
```
```suggestion
	tastes = list("bun" = 2, "dead rat" = 5)
	
```

Contrabang commented on code/modules/food_and_drinks/recipes/recipes_microwave.dm:
```diff
@@ -476,10 +476,141 @@
 		/obj/item/reagent_containers/food/snacks/cheesewedge,
 		/obj/item/reagent_containers/food/snacks/boiledegg,
-		/obj/item/reagent_containers/food/snacks/grown/tomato
+		/obj/item/reagent_containers/food/snacks/tomatoslice
```
Can you mention this in your PR please?

Mitchs98 replied:
Yeah I was gonna, I forgor

Contrabang commented on code/modules/food_and_drinks/recipes/recipes_microwave.dm:
```diff
@@ -476,10 +476,141 @@
+	result = /obj/item/reagent_containers/food/snacks/blt
+
+/datum/recipe/microwave/peanut_butter_jelly
+	reagents = list("cherryjelly" = 5, "peanutbutter" = 5)
```
This is more of a request than a review, but I'd love if we could make these with slime jelly too, like other jelly foods

Mitchs98 replied:
I could do that, sure, it wouldn't be any extra effort really.

Mitchs98 commented:
PB&J can now be made from Slime Jelly as well - per request of Contrabang.

Mitchs98 commented:
ALSO! I forgot to mention this but if someone could help me figure out why Yellow and White Pretty Patties aren't working/overlaying their colors correctly that would be awesome. TG colors theirs the exact some way so I have no idea what's wrong. If it's out of scope for this PR then I'll comment out the recipes until something can be done to fix it.

GDNgit commented on code/modules/food_and_drinks/recipes/recipes_microwave.dm:
```diff
@@ -476,10 +476,149 @@
+
+/datum/recipe/microwave/peanut_butter_jelly/slime
+	reagents = list("slimejelly" = 5, "peanutbutter" = 5)
+	items(
```
```suggestion
	items = list(
```

Mitchs98 replied:
Whoops. Thanks.

Henri215 commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -73,6 +73,95 @@
+	trash = /obj/item/trash/snack_bowl
+	filling_color = "#468C00"
+	bitesize = 3
+	list_reagents = list()
+	tastes = list()
```
No reagents and tastes?

Contrabang commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -136,6 +136,159 @@
+	icon_state = "ratburger"
+	filling_color = "#808080"
+	bitesize = 2
+	list_reagents = list("nutriment"= 5, "protein" = 1)
```
```suggestion
	list_reagents = list("nutriment" = 5, "protein" = 1)
```

ATP-Engineer commented:
will pretty patty ingredients basically be the same as cotton candy using the same juices instead of crayons

Mitchs98 commented:
> will pretty patty ingredients basically be the same as cotton candy using the same juices instead of crayons

Yup. I swapped the ingredients around already!

Also addressed the recent reviews. 

I also split most food items away from food.dmi. If this needs to be its own PR, I understand and can totally swap those changes over. HOWEVER. This is a good thing for the future; not only will the game load less icons when trying to load food(at-least that's how it works right??) there will be fewer conflicts when people are trying to modify anything to do with food.dmi.

Some food items have NOT been split, and this is due to me being unsure of how exactly I should split them. If you have suggestions, I'm all ears.

DT2399 commented:
color burguers!

Henri215 commented on code/modules/food_and_drinks/food/foods/meat.dm:
```diff
@@ -67,12 +68,14 @@
 	desc = "This doesn't look like meat, but your standards aren't <i>that</i> high to begin with."
 	list_reagents = list("nutriment" = 3, "vitamin" = 2, "blood" = 5)
 	filling_color = rgb(150, 0, 0)
+	icon = 'icons/obj/food/meat.dmi'
```
It inherits this from ```/obj/item/reagent_containers/food/snacks/meat``` already, this is the case here and in other places, i suggest taking another look on all of them.

Mitchs98 replied:
I was unaware that setting it for meat would make it inherit all the way down the chain for icon. However, given how the other icon swaps work this would probably only work for soup; which I can in-fact change for soup. I don't think there's any others it would work for other than that due to how things are pathed. I can totally change it for those though, thought I had to path it this way! Woulda saved me a decent chunk of time haha.

Contrabang commented on code/modules/food_and_drinks/food/foods/baked_goods.dm:
```diff
@@ -468,12 +512,14 @@
 /obj/item/reagent_containers/food/snacks/pancake/berry_pancake
 	name = "berry pancake"
 	desc = "A pancake loaded with berries."
+	icon = 'icons/obj/food/breakfast.dmi'
 	icon_state = "berry_pancake"
 	list_reagents = list("nutriment" = 3, "sugar" = 3, "berryjuice" = 3)
 
 /obj/item/reagent_containers/food/snacks/pancake/choc_chip_pancake
 	name = "choc-chip pancake"
 	desc = "A pancake loaded with chocolate chips."
+	icon = 'icons/obj/food/breakfast.dmi'
```
```suggestion
	icon_state = "berry_pancake"
	list_reagents = list("nutriment" = 3, "sugar" = 3, "berryjuice" = 3)

/obj/item/reagent_containers/food/snacks/pancake/choc_chip_pancake
	name = "choc-chip pancake"
	desc = "A pancake loaded with chocolate chips."
```

Contrabang commented on code/modules/food_and_drinks/food/foods/bread.dm:
```diff
@@ -162,6 +178,7 @@
 /obj/item/reagent_containers/food/snacks/baguette
 	name = "baguette"
 	desc = "Bon appetit!"
+	icon = 'icons/obj/food/burgerbread.dmi'
```
Shouldnt this be like, a baked good?

Contrabang commented on code/modules/food_and_drinks/food/foods/meat.dm:
```diff
@@ -369,6 +382,7 @@
 /obj/item/reagent_containers/food/snacks/egg
 	name = "egg"
 	desc = "An egg!"
+	icon = 'icons/obj/food/meat.dmi'
```
Is there no better place to put this? Maybe breakfast.dmi?

Contrabang commented on code/modules/food_and_drinks/food/foods/meat.dm:
```diff
@@ -451,20 +466,23 @@
 /obj/item/reagent_containers/food/snacks/boiledegg
 	name = "boiled egg"
 	desc = "A hard boiled egg."
+	icon = 'icons/obj/food/meat.dmi'
```
ditto

Contrabang commented on code/modules/food_and_drinks/food/foods/meat.dm:
```diff
@@ -451,20 +466,23 @@
 /obj/item/reagent_containers/food/snacks/chocolateegg
 	name = "chocolate egg"
 	desc = "Such sweet, fattening food."
+	icon = 'icons/obj/food/meat.dmi'
```
ditto

Contrabang commented on code/modules/food_and_drinks/food/foods/meat.dm:
```diff
@@ -549,4 +572,5 @@
 	list_reagents = list("nutriment" = 3, "protein" = 10, "bbqsauce" = 10)
 	filling_color = "#FF1C1C"
 	bitesize = 3
+	icon = 'icons/obj/food/meat.dmi'
 	icon_state = "bbqribs"
```
```suggestion
	icon = 'icons/obj/food/meat.dmi'
	icon_state = "bbqribs"
	list_reagents = list("nutriment" = 3, "protein" = 10, "bbqsauce" = 10)
	filling_color = "#FF1C1C"
	bitesize = 3
```

Contrabang commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -66,16 +66,116 @@
+	list_reagents = list("nutriment" = 12, "vitamin" = 6)
+	tastes = list("healthy greens" = 2, "olive dressing" = 2, "feta cheese" = 2, "pita bread" = 1)
+
+/obj/item/reagent_containers/food/snacks/fruitsalad
```
A lot of the salads you're adding have this stuff in common. No need to touch old foods, but could you make all of these salads into a new path? A lot of the cooking stuff like this sucks at inheritence
```suggestion
/obj/item/reagent_containers/food/snacks/salad/fruit
```

Contrabang commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -29,6 +31,7 @@
 /obj/item/reagent_containers/food/snacks/human/burger
```
Unrelated, but I hate this fuckin path

Mitchs98 replied:
Hopefully in a later PR I or someone else can get this to work properly so the path can be killed.

Contrabang commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -130,15 +143,187 @@
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "purple" = 2)
+
+/obj/item/reagent_containers/food/snacks/ppattywhite
```
Could you make the pretty patties use inheritence? Maybe having the white pretty patty be the base because of the default filling color
```suggestion
/obj/item/reagent_containers/food/snacks/ppatty/white
```

Contrabang commented on code/modules/food_and_drinks/recipes/recipes_microwave.dm:
```diff
@@ -476,10 +476,153 @@
+/datum/recipe/microwave/ratburger
+	items = list(
+		/obj/item/reagent_containers/food/snacks/bun,
+		/obj/item/holder/mouse
```
im going to put living mice in the microwave, this is gonna be great

Mitchs98 replied:
Now more than just diona nymphs can receive the microwave treatment.

Mitchs98 commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -63,17 +63,113 @@
 //////////////////////
 
-/obj/item/reagent_containers/food/snacks/aesirsalad
+/obj/item/reagent_containers/food/snacks/salad // Abstract object used for inheritance. Should never spawn. Needed to not break icons.
```
If I moved it up wouldn't that mean every future salad would have a bowl for their trash item? How would I set it so that the melon bowl wouldn't have one?

Contrabang replied:
Yeah, monster meat needs to be changed someday in the future too. Ideally, a generic salad recipee would be good instead of having to move something non-generic to that path.
If you want to set something to not have trash item, you just do `trash = null`

Mitchs98 replied:
Alright I'll make a generic salad out of the unused sprite.

And did figure that out thanks to DG's help, wasn't sure if it was acceptable to just use null like that; but I did, and it's fixed!

Mitchs98 commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -63,17 +63,113 @@
+/obj/item/reagent_containers/food/snacks/salad/aesir
 	name = "aesir salad"
 	desc = "Probably too incredible for mortal men to fully enjoy."
+	icon = 'icons/obj/food/soupsalad.dmi'
```
Missed one, it seems.

Mitchs98 commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -136,18 +143,171 @@
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "purple" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppattyrainbow
```
Not that I disagree, but I'm not quite sure what they would inherit here?

Contrabang replied:
It just makes it better organized imo. We have a lot of shit thats similar, but dont share a path because of old coders. It also makes it easier for getting all of them, as like typesof() or subtypes() of

Mitchs98 replied:
Ahh, alright, fair point and reasonable. Totally don't mind doing it at all!

Mitchs98 commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -3,7 +3,22 @@
 //////////////////////
 
-/obj/item/reagent_containers/food/snacks/brainburger
+/obj/item/reagent_containers/food/snacks/burger // Abstract object used for inheritance. Should never spawn. Needed to not break recipes.
```
I was initially going to do that, however, it broke any recipe that used/required just a plain burger and allowed you to stick any burger into it(which for right now is just the big bite burger though it could end up changing). Hence the line about not breaking recipes. I'd be all for changing it if you have suggestions on how to stop that from occurring. I could make it contain/use maybe the cheeseburger instead, I guess, but I preferred to keep recipes the same for the most part.

DGamerL commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -63,33 +63,101 @@
 	filling_color = "#468C00"
 	bitesize = 3
+	list_reagents = list("nutriment" = 6,"vitamin" = 4)
+	tastes = list("lettuce" = 2)
```
Why would a lone thing in a weighted list have a weight of 2? Could probably make this a normal list

Mitchs98 replied:
As commented on the Discord; every other item I can see for the most part does it this way. I presume it's for custom food weighting I.E you cram it into a sandwich. It's consistent with other food.

DGamerL commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -63,33 +63,101 @@
+	desc = "Your standard fruit salad."
+	icon_state = "fruitsalad"
+	list_reagents = list("nutriment" = 9, "vitamin" = 4)
+	tastes = list("fruit" = 1)
```
Same here, it doesn't hurt much, but I don't think this needs to be a weighted list

DGamerL commented on code/modules/food_and_drinks/food/foods/misc_food.dm:
```diff
@@ -63,33 +63,101 @@
+	icon_state = "melonfruitbowl"
+	trash = null
+	list_reagents = list("nutriment" = 6, "vitamin" = 4)
+	tastes = list("melon" = 1)
```
And here

DGamerL commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -3,7 +3,24 @@
+	filling_color = "#D63C3C"
+	bitesize = 3
+	list_reagents = list("nutriment" = 6, "vitamin" = 1)
+	tastes = list("bun" = 4, "meat" = 1, "the jungle" = 1)
```
Tasting like the jungle seems like something that carried over from monkeyburgers? Not sure what happened there honestly

Mitchs98 replied:
Yeaah. That's actually exactly what happened lol.

DGamerL commented on code/modules/food_and_drinks/food/foods/sandwiches.dm:
```diff
@@ -136,18 +145,171 @@
+	list_reagents = list("nutriment" = 7, "protein" = 1)
+	tastes = list("bun" = 2, "meat" = 2, "purple" = 2)
+
+/obj/item/reagent_containers/food/snacks/burger/ppatty/rainbow
+	name = "rainbow pretty patty"
+	desc = "Taste the rainbow, eat the rainbow."
+	icon_state = "ppatty-rainbow"
+	filling_color = "#0000FF"
+	bitesize = 4
+	list_reagents = list("nutriment" = 14, "protein" = 5)
+	tastes = list("bun" = 2, "meat" = 2, "rainbow" = 5)
```
The other rainbow thing that I can think of (Cotton candy) has omnizine, you could make that a bit consistent?

DGamerL commented on code/modules/food_and_drinks/food/foods/soups.dm:
```diff
@@ -123,6 +124,77 @@
 	. = ..()
 	name = pick("borsch", "bortsch", "borstch", "borsh", "borshch", "borscht")
 
```
Unrelated but this is incredibly funny

DGamerL commented on code/modules/food_and_drinks/food/foods/soups.dm:
```diff
@@ -140,11 +212,39 @@
+	tastes = list("beans" = 1, "cabbage" = 1, "spicy sauce" = 1)
+
+/obj/item/reagent_containers/food/snacks/soup/oatstew
+	name = "Häfmisklhaskl" //häfmiskl = oat (häf from German hafer meaning oat, miskl meaning seed), haskl = stew
```
Bless you

DGamerL reviewed (commented):
Mmmmmm soup
